test/Microsoft.Extensions.ProjectModel.Tests/MsBuild/MsBuildProjectDependencyProviderTests.cs

```diff
@@ -0,0 +1,171 @@
+    <Reference Include = ""xyz.dll"" />
+  </ItemGroup>
+  <Import Project=""$(MSBuildToolsPath)\Microsoft.CSharp.targets"" />
+</Project>
+";
+
+        private const string LibraryProjectTxt = @"
+<Project ToolsVersion=""14.0"" xmlns=""http://schemas.microsoft.com/developer/msbuild/2003"">
+  <Import Project=""$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props"" />
+
+  <PropertyGroup>
+    <RootNamespace>Microsoft.Library</RootNamespace>
+    <ProjectName>Library1</ProjectName>
+    <OutputType>Library</OutputType>
+    <TargetFrameworks>netcoreapp1.0</TargetFrameworks>
+    <OutputPath>bin\$(Configuration)</OutputPath>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <Compile Include=""**\*.cs"" Exclude=""Excluded.cs"" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include=""Microsoft.NETCore.Sdk"">
+      <Version>1.0.0-*</Version>
+    </PackageReference>
+    <PackageReference Include=""Microsoft.NETCore.App"">
+      <Version>1.0.1</Version>
+    </PackageReference>
+  </ItemGroup>
+  <Import Project=""$(MSBuildToolsPath)\Microsoft.CSharp.targets"" />
+</Project>
+";
+        private readonly MsBuildFixture _fixture;
+        private readonly ITestOutputHelper _output;
+
+        public MsBuildProjectDependencyProviderTests(MsBuildFixture fixture, ITestOutputHelper output)
+        {
+            _fixture = fixture;
+            _output = output;
+        }
+
+        [Fact(Skip = "CI doesn't yet have a new enough version of .NET Core SDK")]
+        public void BuildDependenciesForProject()
+        {
+            using (var fileProvider = new TemporaryFileProvider())
+            {
+                Directory.CreateDirectory(Path.Combine(fileProvider.Root, "Root"));
+                Directory.CreateDirectory(Path.Combine(fileProvider.Root, "Library1"));
+                // TODO remove when SDK becomes available on other feeds
+                fileProvider.Add("NuGet.config", NugetConfigTxt);
+
+                // Add Root Project
+                fileProvider.Add($"Root{Path.DirectorySeparatorChar}test.csproj", RootProjectTxt);
+                fileProvider.Add($"Root{Path.DirectorySeparatorChar}One.cs", "public class Abc {}");
+                fileProvider.Add($"Root{Path.DirectorySeparatorChar}Two.cs", "public class Abc2 {}");
+                fileProvider.Add($"Root{Path.DirectorySeparatorChar}Excluded.cs", "public class Abc {}");
+
+                // Add Class Library project
+                fileProvider.Add($"Library1{Path.DirectorySeparatorChar}library.csproj", LibraryProjectTxt);
+                fileProvider.Add($"Library1{Path.DirectorySeparatorChar}Three.cs", "public class Abc3 {}");
+
+                var testContext = _fixture.GetMsBuildContext();
+
+                var muxer = Path.Combine(testContext.ExtensionsPath, "../..", "dotnet.exe");
+                var result = Command
+                    .Create(muxer, new[] { "restore3", Path.Combine(fileProvider.Root, "Library1","Library1.csproj") })
```

Comment:
Is it possible to do restore on a solution file instead of on each project? (I wouldn't be surprised if that doesn't work yet.)